### ISSUE TITLE
Force default value is not taken by ansilbe when rendering template.

### ISSUE THREAD

**Issue by dmilind** [ansible/ansible#45245]
##### SUMMARY

I used a template (abc.yml.j2) to render the file (abc.yml), at first execution ansible placed proper abc.yml file on remote host. Then I made some changes in template and re executed ansible , but nothing has changed in remote abc.yml file. When I explicitely mentioned `force: yes` flag then it got rendered correctly. So as per document `force` flags's default value which is mentioned as `yes` is not default :( 
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

template
##### ANSIBLE VERSION

```paste below
ansible 2.4.2.0
```

##### CONFIGURATION

```paste below
ANSIBLE_SSH_ARGS(/etc/ansible/ansible.cfg) = -o ControlMaster=auto -o ControlPersist=60s
DEFAULT_GATHERING(/etc/ansible/ansible.cfg) = smart
HOST_KEY_CHECKING(/etc/ansible/ansible.cfg) = False
RETRY_FILES_ENABLED(/etc/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT

CentOS Linux release 7.4.1708 (Core)
##### STEPS TO REPRODUCE

```yaml
- name: Update yaml For Beats
  template:
    src: abc.yml.j2
    dest: abc.yml
    owner: root
    group: root
    mode: 0644
    force: yes # Added later 
```

##### EXPECTED RESULTS

After placing `force: yes` flag:

```
TASK [Update yml] ***************************************************************************************************************************************************************************
ok: [x.x.x.x] => (item=abc.yml)

TASK [beats : Start Service] ***********************************************************************************************************************************************************************************

RUNNING HANDLER [beats : Restart Filebeat] *********************************************************************************************************************************************************************

TASK [beats : Restart Filebeat] ********************************************************************************************************************************************************************************
changed: [x.x.x.x]

RUNNING HANDLER [beats : Restart Metricbeat] *******************************************************************************************************************************************************************

TASK [beats : Restart Metricbeat] ******************************************************************************************************************************************************************************
changed: [x.x.x.x]

PLAY RECAP *****************************************************************************************************************************************************************************************************
x.x.x.x.               : ok=7    changed=3    unreachable=0    failed=0
```
##### ACTUAL RESULTS

Before `force: yes` not placed:

```
TASK [beats : Update abc.yml ] ***************************************************************************************************************************************************************************
ok: [x.x.x.x] => (item=abc.yml)

PLAY RECAP *****************************************************************************************************************************************************************************************************
x.x.x.x               : ok=1    changed=0    unreachable=0    failed=0
```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/files/template.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/files/template.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by samdoran** [comment]
I am unable to duplicate this. Is it possible the source template did not get saved, or that the destination already matches the new changes for some reason?

Try running with `--diff --check` to see if Ansible sees any changes need to be made to the file.

```
> ansible --version
ansible 2.4.2.0 (detached HEAD e3a8bf02ac) last updated 2018/09/06 16:23:30 (GMT -400)
  config file = /Users/sdoran/Projects/Vagrant/Ansible Lab/ansible.cfg
  configured module search path = ['/Users/sdoran/Source/ansible/library']
  ansible python module location = /Users/sdoran/Source/ansible/lib/ansible
  executable location = /Users/sdoran/Source/ansible/bin/ansible
  python version = 3.6.5 (default, Jun 21 2018, 15:17:32) [GCC 4.2.1 Compatible Apple LLVM 9.1.0 (clang-902.0.39.2)]
```

```yaml
- name: Local testing playbook
  hosts: localhost
  connection: local
  gather_facts: no
  become: no

  tasks:
    - template:
        src: templates/test.j2
        dest: templates/test.txt
```

```
> ansible-playbook testlocal.yml --diff --check

PLAY [Local testing playbook] **************************************************************

TASK [template] ****************************************************************************
--- before: templates/test.txt
+++ after: /tmp/tmp1r6iky6f/test.j2
@@ -1 +1,2 @@
 line1
+line2

changed: [localhost]

PLAY RECAP *********************************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0
```

needs_info

**Comment by ansibot** [comment]
@dmilind This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@dmilind You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
